secure/storage/src/tests/vault.rs

```diff
@@ -210,61 +308,197 @@ fn test_vault_crypto_policies() {
     exporter_store.sign(CRYPTO_KEY, &message).unwrap_err();
+    // Verify a exporter with another namespace has no permission for the operations
+    let mut exporter_store_with_namespace = VaultStorage::new(
+        dev::test_host(),
+        exporter_token,
+        Some(VAULT_NAMESPACE_1.into()),
+        None,
+        None,
+        true,
+    );
+    exporter_store_with_namespace
+        .export_private_key(CRYPTO_KEY)
+        .unwrap_err();
+    exporter_store_with_namespace
+        .get_public_key(CRYPTO_KEY)
+        .unwrap_err();
+    exporter_store_with_namespace
+        .rotate_key(CRYPTO_KEY)
+        .unwrap_err();
+    exporter_store_with_namespace
+        .sign(CRYPTO_KEY, &message)
+        .unwrap_err();
 
     // Verify noone policy
     let noone_token = storage.create_token(vec![&NOONE]).unwrap();
-    let mut noone_store = VaultStorage::new(dev::test_host(), noone_token, None, None, None, true);
+    let mut noone_store = VaultStorage::new(
+        dev::test_host(),
+        noone_token.clone(),
+        None,
+        None,
+        None,
+        true,
+    );
     noone_store.export_private_key(CRYPTO_KEY).unwrap_err();
     noone_store.get_public_key(CRYPTO_KEY).unwrap_err();
     noone_store.rotate_key(CRYPTO_KEY).unwrap_err();
     noone_store.sign(CRYPTO_KEY, &message).unwrap_err();
+    // Verify a noone with another namespace has no permission for the operations
+    let mut noone_store_with_namespace = VaultStorage::new(
+        dev::test_host(),
+        noone_token,
+        Some(VAULT_NAMESPACE_1.into()),
+        None,
+        None,
+        true,
+    );
+    noone_store_with_namespace
+        .export_private_key(CRYPTO_KEY)
+        .unwrap_err();
+    noone_store_with_namespace
+        .get_public_key(CRYPTO_KEY)
+        .unwrap_err();
+    noone_store_with_namespace
+        .rotate_key(CRYPTO_KEY)
+        .unwrap_err();
+    noone_store_with_namespace
+        .sign(CRYPTO_KEY, &message)
+        .unwrap_err();
 
     // Verify reader policy
     let reader_token = storage.create_token(vec![&READER]).unwrap();
-    let mut reader_store =
-        VaultStorage::new(dev::test_host(), reader_token, None, None, None, true);
+    let mut reader_store = VaultStorage::new(
+        dev::test_host(),
+        reader_token.clone(),
+        None,
+        None,
+        None,
+        true,
+    );
     reader_store.export_private_key(CRYPTO_KEY).unwrap_err();
     assert_eq!(
         reader_store.get_public_key(CRYPTO_KEY).unwrap().public_key,
         pubkey
     );
     reader_store.rotate_key(CRYPTO_KEY).unwrap_err();
     reader_store.sign(CRYPTO_KEY, &message).unwrap_err();
+    // Verify a reader with another namespace has no permission for the operations
+    let mut reader_store_with_namespace = VaultStorage::new(
+        dev::test_host(),
+        reader_token,
+        Some(VAULT_NAMESPACE_1.into()),
+        None,
+        None,
+        true,
+    );
+    reader_store_with_namespace
```

Comment:
I'd just check that "get_public_key" doesn't work here. I'm not sure we gain much by checking all the variants.